src/app/shared/http-services/get-group-users-progress.service.ts

```diff
@@ -0,0 +1,56 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { appConfig } from 'src/app/shared/helpers/config';
+
+interface RawGroupUsersProgress {
+  group_id: string,
+  hints_requested: number,
+  item_id: string,
+  latest_activity_at: string|null,
+  score: number,
+  submissions: number,
+  time_spent: number,
+  validated: boolean,
+}
+
+export interface GroupUsersProgress {
+  groupId: string,
+  hintsRequested: number,
+  itemId: string,
+  latestActivityAt: Date|null,
+  score: number,
+  submissions: number,
+  timeSpent: number,
+  validated: boolean,
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class GetGroupUsersProgressService {
```

Comment:
Maybe you should group all the progress services in a single file.

Reply:
for now I just have one but I'll do this with the others.